openreview/invitations/templates.py

```diff
@@ -232,4 +161,42 @@ def __init__(self, name, conference_id, duedate = 0,
         super(Comment, self).__init__(**self.inv_params)
 
     def add_process(self, process):
-        self.process = process.render()
\ No newline at end of file
+        self.process = process.render()
+
+def fill_template(template_string, paper_params):
+    pattern = '|'.join(['<{}>'.format(field) for field, value in paper_params.iteritems()])
+    matches = re.findall(pattern, template_string)
+    for match in matches:
+        discovered_field = re.sub('<|>', '', match)
+        template_string = template_string.replace(match, str(paper_params[discovered_field]))
+        print "    new value: ", template_string
+    return template_string
+
+def generate_invitation(invitation_template, paper):
+    params = copy.deepcopy(invitation_template)
+    reply = params.pop('reply')
+    paper_params = paper.to_json()
+
+    def fill_list_or_str(value, paper_params=paper_params):
+        if type(value) == list:
+            return [fill_template(v, paper_params) for v in value]
```

Comment:
you are defining an inner function that is calling to an outer function, is this ok?

Why do you define an inner function?

Reply:
another good point... I'm not sure why this is an inner function, it could just as easily be an outer function